app/scripts/services/applicationGenerator.js

```diff
@@ -395,13 +349,71 @@ angular.module("openshiftConsole")
           selector: {
             deploymentconfig: input.name
           },
-          ports: _.map(ports, scope.getServicePort)
+          ports: _.map(ports, getServicePort)
         }
       };
 
       return service;
     };
 
-    return scope;
-  }
-);
+    /**
+     * Generate resource definitions to support the given input
+     * @param {type} input
+     * @returns Hash of resource definitions
+     */
+    var generate = function(input){
+      var ports = parsePorts(input.image);
+
+      //augment labels
+      input.labels.app = input.name;
+      input.annotations["openshift.io/generated-by"] = "OpenShiftWebConsole";
+
+      var imageSpec;
+      if(input.buildConfig.sourceUrl !== null){
+        imageSpec = {
+          name: input.name,
+          tag: "latest",
+          kind: "ImageStreamTag",
+          toString: function(){
+            return this.name + ":" + this.tag;
+          }
+        };
+      }
+
+      var resources = {
+        imageStream: _generateImageStream(input),
+        buildConfig: _generateBuildConfig(input, imageSpec, input.labels),
+        deploymentConfig: _generateDeploymentConfig(input, imageSpec, ports, input.labels)
+      };
+
+      if (input.scaling.autoscale) {
+        resources.hpa = _generateHPA(input, resources.deploymentConfig);
+      }
+
+      var service = _generateService(input, input.name, ports);
+      if (service) {
+        resources.service = service;
+
+        // Only attempt to generate a route if there is a service.
+        resources.route = _generateRoute(input, input.name, resources.service.metadata.name);
+      }
+
+      return resources;
+    };
+
+    return {
+      createRoute: createRoute,
+      generate: generate,
+      parsePorts: parsePorts,
+      generateSecret: generateSecret,
+      getServicePort: getServicePort,
+      // pseudo private
```

Comment:
right... we have tests already written so I didn't want to just remove the tests.  Perhaps they dont even need to be semi-private?